src/dbnode/network/server/tchannelthrift/node/service.go

```diff
@@ -256,22 +263,49 @@ func (s *service) Bootstrapped(ctx thrift.Context) (*rpc.NodeBootstrappedResult_
 }
 
 func (s *service) Query(tctx thrift.Context, req *rpc.QueryRequest) (*rpc.QueryResult_, error) {
+	ctx := tchannelthrift.Context(tctx)
+
+	var (
+		sp    opentracing.Span
+		spCtx stdctx.Context
+	)
+
+	goCtx, exists := ctx.GoContext()
+	if exists {
+		sp, spCtx = opentracingutil.StartSpanFromContext(goCtx, querySpanID)
+		sp.LogFields(
+			opentracinglog.String("query", req.Query.String()),
+			opentracinglog.String("namespace", req.NameSpace),
+			opentracingutil.Time("start", time.Unix(0, req.RangeStart)),
+			opentracingutil.Time("end", time.Unix(0, req.RangeStart)),
+		)
+
+		ctx.SetGoContext(spCtx)
+		defer sp.Finish()
```

Comment:
This finish only applies to the jaeger trace right?

Reply:
Yes I believe so.